src/ScheduledTasks/Client/ScheduledTaskClientImpl.cs

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.DurableTask.Client;
+using Microsoft.DurableTask.Client.Entities;
+using Microsoft.DurableTask.Entities;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DurableTask.ScheduledTasks;
+
+/// <summary>
+/// Client for managing scheduled tasks in a Durable Task application.
+/// </summary>
+#pragma warning disable CA1711 // Identifiers should not have incorrect suffix
+public class ScheduledTaskClientImpl(DurableTaskClient durableTaskClient, ILogger logger) : ScheduledTaskClient
+#pragma warning restore CA1711 // Identifiers should not have incorrect suffix
+{
+    readonly DurableTaskClient durableTaskClient = Check.NotNull(durableTaskClient, nameof(durableTaskClient));
+    readonly ILogger logger = Check.NotNull(logger, nameof(logger));
+
+    /// <inheritdoc/>
+    public override async Task<ScheduleClient> CreateScheduleAsync(ScheduleCreationOptions creationOptions, CancellationToken cancellation = default)
+    {
+        Check.NotNull(creationOptions, nameof(creationOptions));
+        this.logger.ClientCreatingSchedule(creationOptions);
+
+        try
+        {
+            string scheduleId = creationOptions.ScheduleId;
+            EntityInstanceId entityId = new EntityInstanceId(nameof(Schedule), scheduleId);
+
+            // Call the orchestrator to create the schedule
+            ScheduleOperationRequest request = new ScheduleOperationRequest(entityId, nameof(Schedule.CreateSchedule), creationOptions);
+            string instanceId = await this.durableTaskClient.ScheduleNewOrchestrationInstanceAsync(
+                new TaskName(nameof(ExecuteScheduleOperationOrchestrator)),
+                request,
+                cancellation);
+
+            // Wait for the orchestration to complete
+            OrchestrationMetadata state = await this.durableTaskClient.WaitForInstanceCompletionAsync(instanceId, true, cancellation);
+
+            if (state.RuntimeStatus != OrchestrationRuntimeStatus.Completed)
+            {
+                throw new InvalidOperationException($"Failed to create schedule '{scheduleId}': {state.FailureDetails?.ErrorMessage ?? string.Empty}");
+            }
+
+            // Return a handle to the schedule
+            return new ScheduleClientImpl(this.durableTaskClient, scheduleId, this.logger);
+        }
+        catch (OperationCanceledException) when (cancellation.IsCancellationRequested)
+        {
+            // the operation was cancelled as requested. No need to log this.
+            throw;
+        }
+        catch (Exception ex)
+        {
+            this.logger.ClientError(
+                nameof(this.CreateScheduleAsync),
+                creationOptions.ScheduleId,
+                ex);
+
+            throw;
+        }
+    }
+
+    /// <inheritdoc/>
+    public override async Task<ScheduleDescription?> GetScheduleAsync(string scheduleId, CancellationToken cancellation = default)
```

Comment:
should not duplicate implementation code between ScheduleClientImpl.cs and ScheduledTaskClientImpl.cs